project/SparkInstaller.scala

```diff
@@ -0,0 +1,172 @@
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{Path, Paths, _}
+
+import sbt.{IO, _}
+
+import scala.collection.JavaConversions._
+import scala.collection.concurrent.TrieMap
+import scala.xml.{Elem, XML}
+
+object SparkInstaller {
+
+  private val attempts = TrieMap[(String, String), Unit]()
+
+  private def isGitVersion(providedVersion: String): Boolean = {
+    val GitTagPattern = """v[0-9].+""".r
+    val GitBranchPattern = """(master|branch).*\-SNAPSHOT""".r
+
+    providedVersion match {
+      case GitTagPattern() => true
+      case GitBranchPattern(_) => true
+      case _ => false
+    }
+  }
+
+  private object StdProcessLogger extends sbt.ProcessLogger {
```

Comment:
This is used for super-nice Scala process invocations with `!`. I wanted to redirect output streams to the current stdout and stderr, especially for long running tasks, like `mvn install`.